# A regression introduced in 2.2.33 (which is also included in the latest 2.2.34)
# causes gpgconf to always start in verbose mode, no matter whether or not the
# --verbose mode is passed. It's also no longer possible to disable verbosity
# with --no-verbose.
# The following code is taken from the 2.3.x code base which doesn't include
# this bug.
#
# A ticket in the GnuPG bug tracker was created:
# https://dev.gnupg.org/T5874

diff --git a/tools/gpgconf-comp.c b/tools/gpgconf-comp.c
index ad3e8d729..4aab171f5 100644
--- a/tools/gpgconf-comp.c
+++ b/tools/gpgconf-comp.c
@@ -1914,8 +1914,10 @@ retrieve_options_from_program (gc_component_id_t component, int only_installed)
   pargs.flags = (ARGPARSE_FLAG_KEEP
                  | ARGPARSE_FLAG_SYS
                  | ARGPARSE_FLAG_USER
-                 | ARGPARSE_FLAG_WITHATTR
-                 | ARGPARSE_FLAG_VERBOSE);
+                 | ARGPARSE_FLAG_WITHATTR);
+
+  if (opt.verbose)
+      pargs.flags |= ARGPARSE_FLAG_VERBOSE;
 
   while (gnupg_argparser (&pargs, opt_table, config_name))
     {
